helm/provider_test.go

```diff
@@ -24,82 +26,163 @@ const (
-		log.Printf("[ERROR] Failed to create new temporary directory for use as helm home: %s", err)
-	}
 }
 
+// 	var err error
+// 	testAccHelmHome, err = ioutil.TempDir("", "terraform-acc-test-helm-")
+// 	if err != nil {
+// 		log.Printf("[ERROR] Failed to create new temporary directory for use as helm home: %s", err)
+// 	}
+// }
+
 func TestProvider(t *testing.T) {
 	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
 		t.Fatalf("err: %s", err)
 	}
 }
 
-func TestAccProviderHelmInitEnabled(t *testing.T) {
-	err := testAccProviderHelmInit(t, true)
+// func TestAccProviderHelmInitEnabled(t *testing.T) {
+// 	err := testAccProviderHelmInit(t, true)
+// 	if err != nil {
+// 		t.Fatal("helm home should have been initialized", err)
+// 	}
+// }
+
+// func TestAccProviderHelmInitDisabled(t *testing.T) {
+// 	err := testAccProviderHelmInit(t, false)
+// 	if err != io.EOF {
+// 		t.Fatal("helm home should not have been initialized", err)
+// 	}
+// }
+
+// func testAccProviderHelmInit(t *testing.T, enabled bool) (err error) {
+// 	if os.Getenv(resource.TestEnvVar) == "" {
+// 		t.Skip(fmt.Sprintf(
+// 			"Acceptance tests skipped unless env '%s' set", resource.TestEnvVar))
+// 		return
+// 	}
+
+// 	helmHome, err := ioutil.TempDir("", "terraform-acc-test-helm-")
+// 	if err != nil {
+// 		t.Fatalf("Failed to create new temporary directory for use as helm home: %s", err)
+// 	}
+// 	defer os.RemoveAll(helmHome)
+
+// 	log.Printf("[INFO] Test: Using %s as helm home", helmHome)
+
+// 	testProvider := Provider().(*schema.Provider)
+// 	err = testProvider.Configure(terraform.NewResourceConfigRaw(map[string]interface{}{
+// 		"home":           helmHome,
+// 		"init_helm_home": enabled,
+// 	}))
+// 	if err != nil {
+// 		t.Fatal(err)
+// 	}
+
+// 	f, err := os.Open(helmHome)
+// 	defer f.Close()
+// 	if err != nil {
+// 		t.Fatal("Failed to access helm home directory")
+// 	}
+
+// 	_, err = f.Readdirnames(1)
+// 	return
+// }
+
+func testAccPreCheck(t *testing.T, namespace string) {
+	err := setK8Client()
+
 	if err != nil {
-		t.Fatal("helm home should have been initialized", err)
+		t.Fatal(err)
 	}
-}
 
-func TestAccProviderHelmInitDisabled(t *testing.T) {
-	err := testAccProviderHelmInit(t, false)
-	if err != io.EOF {
-		t.Fatal("helm home should not have been initialized", err)
+	err = testAccProvider.Configure(terraform.NewResourceConfigRaw(nil))
+
+	if err != nil {
+		t.Fatal(err)
 	}
-}
 
-func testAccProviderHelmInit(t *testing.T, enabled bool) (err error) {
-	if os.Getenv(resource.TestEnvVar) == "" {
-		t.Skip(fmt.Sprintf(
-			"Acceptance tests skipped unless env '%s' set", resource.TestEnvVar))
-		return
+	if namespace != "" {
+		createNamespace(t, namespace)
 	}
 
-	helmHome, err := ioutil.TempDir("", "terraform-acc-test-helm-")
-	if err != nil {
-		t.Fatalf("Failed to create new temporary directory for use as helm home: %s", err)
+	os.Setenv("HELM_REPOSITORY_CONFIG", "/Users/amell/Library/Preferences/helm/repositories.yaml")
```

Comment:
I could use some advice here. Not sure how these should be set.

Reply:
Look at https://golang.org/pkg/os/#UserConfigDir

Reply:
Also, chek helm docs -- they may use XDG not only on Linux/BSD, but also on Mac/Windows?